server/src/main/java/org/opensearch/rest/action/search/RestSearchAction.java

```diff
@@ -134,6 +136,15 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
             parser -> parseSearchRequest(searchRequest, request, parser, client.getNamedWriteableRegistry(), setSize)
         );
 
+        if (FeatureFlags.isEnabled(FeatureFlags.STREAM_TRANSPORT)) {
```

Comment:
Can we have a different feature flag for stream search? and check for both here? There can be a case where stream transport is needed by another feature but not stream search API

Reply:
So for someone using REST API, a query param `stream=true` is needed to enable stream search.
It's not written as a feature flag, but does the same thing. Basically stream search API is still disabled by default.
The code related to it is also labeled as `@Experimental` because we may not need it one day.